File: Solutions/1423. Maximum Points You Can Obtain from Cards/1423. Maximum Points You Can Obtain from Cards.go
func maxScore(cardPoints []int, k int) int {
    n,lsum:=len(cardPoints),0
    for i:=0;i<k;i+=1{
        lsum+=cardPoints[i]
    }
    mx,rsum:=lsum,0
    for i:=0;i<k;i+=1{
        rsum+=cardPoints[n-1-i]
        lsum-=cardPoints[k-1-i]
        mx=max(mx,lsum+rsum)
    }
    return mx
}

func max (i,j int) int{
    if i>j{
        return i
    }
    return j
}

File: Solutions/1423. Maximum Points You Can Obtain from Cards/1423. Maximum Points You Can Obtain from Cards.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int n=cardPoints.size(),lsum=0;
        for(int i=0;i<k;++i){
            lsum+=cardPoints[i];
        }
        int mx=lsum;
        for(int rsum=0,i=0;i<k;++i){
            rsum+=cardPoints[n-i-1];
            lsum-=cardPoints[k-1-i];
            mx=max(mx,lsum+rsum);
        }
        return mx;
    }
};